src/main/scala/com/newlibertie/pollster/impl/Poll.scala

```diff
@@ -1,25 +1,28 @@
 package com.newlibertie.pollster.impl
 
 import java.math.BigInteger
+import java.util.Date
 
 import com.newlibertie.pollster.DataAdapter
 import net.liftweb.json._
 
+import scala.collection.mutable
+
 object Poll {
+  implicit val formats: DefaultFormats.type = DefaultFormats
 
   def apply(params: PollParameters): Poll = new Poll(params)
 
   def apply(pollDetails:String): Poll = {
-    implicit val formats: DefaultFormats.type = DefaultFormats
     val jValue = parse(pollDetails)
     val pollParameters = jValue.extract[PollParameters]
     val pollJsonMap = jValue.values.asInstanceOf[Map[String, String]]
     val cryptographicParameters = if(  // string did not prescribe existing crypto params, then create new
       !pollJsonMap.contains("p") ||
-      !pollJsonMap.contains("g") ||
-      !pollJsonMap.contains("s")){
-        new CryptographicParameters()
-      }
+        !pollJsonMap.contains("g") ||
+        !pollJsonMap.contains("s")){
+      new CryptographicParameters()
```

Comment:
I know I had implemented the above apply.  But now it feels that two paths within the function are unsafe to some extent

Can you please add a log statement so we will always know which branch was taken